src/test/java/com/wiley/autotest/framework/pages/RepeaterPage.java

```diff
@@ -0,0 +1,42 @@
+package com.wiley.autotest.framework.pages;
+
+import com.wiley.autotest.actions.RepeatableAction;
+import com.wiley.autotest.selenium.context.AbstractPage;
+import com.wiley.autotest.selenium.context.SearchStrategy;
+import org.openqa.selenium.By;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by shekhavtsov on 29/09/2017.
+ */
+@Component
+public class RepeaterPage extends AbstractPage {
+
+    public RepeaterPage negativeCheck() {
+        new RepeatableAction(this::clickOnFailure).perform();
+        return this;
+    }
+
+    public RepeaterPage positiveCheck() {
+        new RepeatableAction(this::clickOnSuccess).perform();
+        return this;
+    }
+
+    private boolean clickOnFailure() {
+        clickDiv("[id='testDiv1']");
+        return divIsNull();
+    }
+
+    private boolean clickOnSuccess() {
+        clickDiv("[id='testDiv2']");
+        return divIsNull();
+    }
+
+    private void clickDiv(String locator) {
+        element(By.cssSelector(locator)).click();
+    }
+
+    private boolean divIsNull() {
```

Comment:
1. typical java naming style for such methods would be isDivNull?
2. what "div is null" means for the application? I think we usually use isDisplayed isAbsent - lets stick to common style

Reply:
renamed to isElementDisplayed